lib/rules/unit-no-unknown/index.js

```diff
@@ -62,10 +63,42 @@ const rule = function(actual, options) {
 					return;
 				}
 
-				if (keywordSets.units.has(unit.toLowerCase())) {
+				if (keywordSets.units.has(unit.toLowerCase()) && unit.toLowerCase() !== 'x') {
 					return;
 				}
 
+				if (unit.toLowerCase() === 'x') {
+					if (
+						node.type === 'atrule' &&
+						node.name === 'media' &&
+						node.params.toLowerCase().includes('resolution')
+					) {
+						let ignoreUnit = false;
+
+						mediaParser(node.params).walk((mediaNode, i, mediaNodes) => {
+							if (
+								mediaNode.value.toLowerCase().includes('resolution') &&
+								_.last(mediaNodes).sourceIndex === valueNode.sourceIndex
+							) {
+								ignoreUnit = true;
+
+								return false;
+							}
+						});
+
+						if (ignoreUnit) {
+							return;
+						}
+					}
+
+					if (
+						node.type === 'decl' &&
+						(node.prop.toLowerCase() === 'image-resolution' || /^image-set/i.test(node.value))
+					) {
```

Comment:
I think we can parse decl here too, because you can use `url("../../media/examples/star.png") left 20x / 15% 60% repeat-x`

Reply:
Nice case. Updated the logic.